src/portable/portable.c

```diff
@@ -1025,10 +1059,16 @@ static int install_chroot_dropin(
                                         return -ENOMEM;
         }
 
-        r = write_string_file(dropin, text, WRITE_STRING_FILE_CREATE|WRITE_STRING_FILE_ATOMIC);
+        r = write_string_file(dropin, text, WRITE_STRING_FILE_CREATE|WRITE_STRING_FILE_ATOMIC|WRITE_STRING_FILE_SYNC);
         if (r < 0)
                 return log_debug_errno(r, "Failed to write '%s': %m", dropin);
 
+        /* Generated on-the-fly so we won't have signature validation, but can still enforce
+         * integrity checks. */
+        r = fsverity_enable(dropin, NULL, 0);
+        if (!IN_SET(r, 0, -EOPNOTSUPP, -ENOTTY))
+                log_debug_errno(r, "Failed to enable fs-verity on '%s', ignoring: %m", dropin);
```

Comment:
maybe make this a new flag to write_string_file() itself?

going via path here kinda sucks, this should be done via fd.

Reply:
we are using the path for everything else here, with the strictness of fd requirements for the ioctl it's not really worth it

Reply:
it doesn't help much given we need to reopen, but added anyway